src/actions/appActions.js

```diff
@@ -19,19 +19,18 @@ export function loadApps(clusterId) {
     const nodePoolsClusters = getState().entities.clusters.nodePoolsClusters;
     const isNodePoolsCluster = nodePoolsClusters.includes(clusterId);
 
-    let getClusterAppsFunc;
+    let getClusterApps = appsApi.getClusterAppsV4;
+
     if (isNodePoolsCluster) {
-      getClusterAppsFunc = appsApi.getClusterAppsV5.bind(appsApi);
-    } else {
-      getClusterAppsFunc = appsApi.getClusterAppsV4.bind(appsApi);
+      getClusterApps = appsApi.getClusterAppsV5;
     }
 
     dispatch({
       type: types.CLUSTER_LOAD_APPS,
       clusterId,
     });
 
-    return getClusterAppsFunc(clusterId)
+    return getClusterApps(appsApi, clusterId)
```

Comment:
Does this refactoring work? I'll test it in a bit. But based on my understanding, I think it shouldn't work.

The first argument to `getClusterAppsV4` or `V5` is the `clusterId`.

Before I was getting the function and using `bind` to keep the executing scope (`this`) of the function the same. Now it's being called with the wrong first argument, and it's scope will be `undefined` I think.

Reply:
Looks like I was too excited to go on holiday, don't know what I was thinking about 😆